src/spec.c

```diff
@@ -1246,16 +1247,35 @@ void CleanPool_ThreadPoolDestroy() {
   }
 }
 
-size_t CleanPool_WorkingThreadCount() {
-  if (cleanPool) {
-    return redisearch_thpool_num_threads_working(cleanPool);
-  }
+size_t getPendingIndexDrop() {
+  return __atomic_load_n(&pendingIndexDropCount_g, __ATOMIC_RELAXED);
+}
+
+size_t addPendingIndexDrop() {
+  __atomic_fetch_add(&pendingIndexDropCount_g, 1, __ATOMIC_RELAXED);
+}
+
+void removePendingIndexDrop() {
+  uint16_t current = __atomic_load_n(&pendingIndexDropCount_g, __ATOMIC_RELAXED);
+  do {
+    if (current == 0) {
+      return;
+    }
+  } while (!__atomic_compare_exchange_n(&pendingIndexDropCount_g, &current, current - 1, 0, 0, 0));
+
+}
+
+size_t CleanInProgressOrPending() {
+  return getPendingIndexDrop() || (cleanPool && !redisearch_thpool_is_idle(cleanPool));
 }
 
 /*
  * Free resources of unlinked index spec
  */
 static void IndexSpec_FreeUnlinkedData(IndexSpec *spec) {
+
+  removePendingIndexDrop();
```

Comment:
consider adding this at the end of the function, after the spec is done being freed